### ISSUE TITLE
Unparse skips columns

### ISSUE THREAD

**Issue by silvestreh** [mholt/PapaParse#609]
Looks like unparsing a JSON array ends up in missing columns if the first `Object` in the array is missing properties that other `Object`s in the same `Array` do have.

This `Array` is correctly turned into CSV:

```javascript
[
  {
    date: '2018-12-20T17:04:41.446Z',
    amount: '30.03',
    metadata_customerEmail: 'someone@domain.com'
  },
  {
    date: '2018-12-20T17:01:30.434Z',
    amount: '13.06',
    metadata_customerEmail: 'someone@domain.com'
  },
  {
    date: '2018-12-20T16:49:55.630Z',
    amount: '31.33',
    metadata_customerEmail: 'someone@domain.com'
  },
  {
    date: '2018-12-20T16:33:50.121Z',
    amount: '29'
  }
]
```

This is the output:

```csv
date,amount,metadata_customerEmail
2018-12-20T17:04:41.446Z,30.03,someone@domain.com
2018-12-20T17:01:30.434Z,13.06,someone@domain.com
2018-12-20T16:49:55.630Z,31.33,someone@domain.com
2018-12-20T16:33:50.121Z,29,
```

Now, if you `sort` that array so the oldest entry (the one without `metadata_customerEmail`) is at `index` 0, then you end up with this CSV:

```csv
date,amount
2018-12-20T16:33:50.121Z,29
2018-12-20T16:49:55.630Z,31.33
2018-12-20T17:01:30.434Z,13.06
2018-12-20T17:04:41.446Z,30.03
```

**Comment by silvestreh** [comment]
Here's a thought: maybe Papa could make a first pass to determine what the columns should be and then a second one to grab all the data?

**Comment by pokoli** [comment]
I don't think we should double pass the results as this will be a performance issue but you may probably do it before passing the values to paparse. This way you can perform any additional checks or whatever you need. 

Do you have some suggestion about how to improve the unparse of empty values? For me the current behavior is correct.

**Comment by MonkeyDZeke** [comment]
I would suggest we add a config variable for unparse called something like `columns` which accepts either an array (`['key_1', 'key_2', 'key_3']`) or an object (`{old_key_1: 'new_key_1', old_key_2: 'new_key_2', old_key_3: 'new_key_3'}`) so that the user can tell the unparser what to expect (and conveniently alter the resulting header row if desired).

Thoughts?

**Comment by dboskovic** [comment]
I agree with @pokoli that this is something that should happen outside of this library. I think the `columns` option with an array of keys passed to unparse would be valuable, both in determining output order and in solving this problem. Don't think we should do the old/new mapping though, that feels bloaty.

**Comment by MonkeyDZeke** [comment]
Makes sense. The resulting header row can be edited afterwards easily enough anyway.

**Comment by pokoli** [comment]
Fixed on #632
